src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorOverloadResolution.cs

```diff
@@ -809,47 +872,29 @@ private static LiftingResult UserDefinedBinaryOperatorCanBeLifted(TypeSymbol lef
             // SPEC: better than all other function members, then the function member invocation is ambiguous and a binding-time 
             // SPEC: error occurs.
 
-            // UNDONE: This is a naive quadratic algorithm; there is a linear algorithm that works. Consider using it.
             var candidates = result.Results;
-            for (int i = 1; i < candidates.Count; ++i)
-            {
-                if (candidates[i].Kind != OperatorAnalysisResultKind.Applicable)
-                {
-                    continue;
-                }
+            var worse = OperatorOverloadResolution(
+                candidates.SelectAsArray(x => (x.Kind, x.Signature)),
```

Comment:
I think the duplication was slightly better than this, let alone using another delegate to access (Kind Signature), since everything else is also duplicated on their own struct types, my vote is to revert this change.